assured-workloads-example/kms.tf

```
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  encryption_keys_project_id = one(
+    [for setting in google_assured_workloads_workload.primary.resource_settings : setting.resource_id
+    if setting.resource_type == "ENCRYPTION_KEYS_PROJECT"]
+  )
+  keyring_id = one(
+    [for setting in google_assured_workloads_workload.primary.resource_settings : setting.resource_id
+    if setting.resource_type == "KEYRING"]
+  )
+}
```

I'm not sure if I followed the idea here. Why can't we do the opposite, and use the `locals` values in both `google_assured_workloads_workload` and `google_kms_crypto_key` resources?

PS: Maybe it is a good idea to append the random suffix into the keyring's name.
PS 2: Since we are using `random_string.id.result` multiple times, I would make a local of it and use `local.default_suffix` instead

```suggestion
locals {
  encryption_keys_project_id = "${var.aw_base_id}-kms-${random_string.id.result}"
  keyring_id = "${var.aw_base_id}-keyring-${random_string.id.result}""
}
```

---

Great! Done.